strings.cc

```diff
@@ -1,23 +1,39 @@
 #include <iostream>
 #include <fstream>
 #include <vector>
-#include <assert.h>
+#include <cassert>
+#include <algorithm>
+
+static const std::string DICTIONARY_PATH = "/usr/share/dict/words";
+
+/**
+ * @brief Return true if the string is a palindrome, false otherwise.
+ * @details Return true if the string is a palindrome. No verification is done
+ * to determine if the input string is a word (e.g. "N" is a palindrome, but is
+ * "N" a word?). All characters in the string are treated as part of the word; 
+ * e.g. "a man a plan a canal panama" is a palindromic sentence, but only
+ * after adjusting the spacing.
+ * 
+ * The solution is O(N) in the number of characters. It uses std::equal to 
+ * compare the first half of the word to the second half in reverse order.
+ * @see std::equal
+ */
+inline bool isPalindromic(const std::string& word){
+    return std::equal(word.begin(), word.begin()+word.size()/2, word.rbegin());
```

Comment:
Interesting: if `word.size()` is odd, there is no need to check the middle letter, and this code doesn't check it.